### PR TITLE
DEV-1051 saving annotations

### PR DESCRIPTION
**Pull request by scottharvey**
This PR add the ability to serialize and deserialze a single highlight.

Related PR: https://github.com/TutoringAustralasia/eureka/pull/3235
Related Card: https://yourtutor.atlassian.net/browse/DEV-1051

### CODE REVIEW

**Comment by scottharvey** [comment]
@abrom Can you review this please 🙏

File: src/TextHighlighter.js [hunk 1]
```diff
@@ -933,6 +969,36 @@
         return JSON.stringify(hlDescriptors);
     };
 
+    TextHighlighter.prototype.deserializeHighlight = function (hl) {
+        var self = this;
+        hl.path = hl.path.split(':');
+        var elIndex = hl.path.pop();
+        var node = self.el;
+        var hlNode, highlight, idx;
```
Comment (abrom) [inline src/TextHighlighter.js]: `highlight` var isn't used

File: src/TextHighlighter.js [hunk 2]
```diff
@@ -876,6 +876,42 @@
         return el && el.nodeType === NODE_TYPE.ELEMENT_NODE && el.hasAttribute(DATA_ATTR);
     };
 
+    TextHighlighter.prototype.serializeHighlight = function (highlight) {
+        var refEl = this.el;
+
+        function getElementPath(el, refElement) {
```
Comment (abrom) [inline src/TextHighlighter.js]: thoughts on pulling this out into a shared private method?

**Review by abrom — APPROVED** [review]
Looks good @scottharvey 👍 

Few minor suggestions but no show stoppers. There could be the option to try reuse the (de)serialiser functions in the pluralised versions but given the different structures might not be worth it..

**Comment by scottharvey** [comment]
@abrom I've made a couple of changes based on your comments. Let me know if you are still happy for this to be merged.

**Review by abrom — APPROVED** [review]
Do iiiit 👍

**Pull request merged**